src/commands/restart.js

```diff
@@ -0,0 +1,81 @@
+const { Listr } = require('listr2');
+
+const { flags: flagTypes } = require('@oclif/command');
+
+const BaseCommand = require('../oclif/command/BaseCommand');
+
+const MuteOneLineError = require('../oclif/errors/MuteOneLineError');
+
+class RestartCommand extends BaseCommand {
+  /**
+   * @param {Object} args
+   * @param {Object} flags
+   * @param {DockerCompose} dockerCompose
+   * @param {startNodeTask} startNodeTask
+   * @param {Config} config
+   * @return {Promise<void>}
+   */
+  async runWithDependencies(
+    args,
+    {
+      update: isUpdate,
+      'drive-image-build-path': driveImageBuildPath,
+      'dapi-image-build-path': dapiImageBuildPath,
+    },
+    dockerCompose,
+    startNodeTask,
+    config,
+  ) {
+    const isMinerEnabled = config.get('core.miner.enable');
+    const isMasternode = config.get('core.masternode.enable');
+
+    const tasks = new Listr(
+      [
+        {
```

Comment:
I would check that node is running otherwise the restart command will start the node even if it’s stopped.

Reply:
Isn't this the behaviour we want? In my mind this function is used after fixing something in the config that is causing the node to run incorrectly, e.g. in stopped or restarting state. If we do want to check it is running, which service should we check? All of them?

Reply:
Hm... I got used if you try to restart stopped thing you get an error that the service is not started. Is not it? You can use `isServiceRunning` without arguments. But probably, it’s better to put in `startNodeTask`.

Reply:
I don't understand your logic here. If we check the services are stopped in `startNodeTask` then in the `restart` command the stop node task will have already completed, so it will always evaluate `true`.

In general, during testing I often run into cases where a service is restarting because of a config problem. All our services are configured with `restart: unless-stopped` in docker-compose, so the only possible states are `running` or `restarting`. If I was to run `mn restart` in this case it would fail if we check for started services first, because the service is always restarting. Which means I am back to typing `mn stop` then `mn start` again anyway. Right?

Reply:
I added a step in `startNodeTask` to check for stopped services, but the `this.error` syntax does not work here, including for the other error checks in this function. It returns `this.error is not a function`. I still don't understand how errors should be handled with all the dependencies, so I don't know how to fix this.

Reply:
Yeah, if service is running (containers are present) then you should be able to use restart.

You just calling `this.error` in wrong context. It's a method of Oclif command, not `Listr`. For Listr task you just need to throw an error (`throw new Error('Node is not started')`);